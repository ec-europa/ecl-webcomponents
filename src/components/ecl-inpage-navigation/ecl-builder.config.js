const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-inpage-navigation/inpage-navigation.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-inpage-navigation-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-inpage-navigation-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-inpage-navigation-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-inpage-navigation-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-inpage-navigation-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
