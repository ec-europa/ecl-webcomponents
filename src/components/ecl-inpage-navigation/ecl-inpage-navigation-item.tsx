import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-inpage-navigation-item',
  shadow: false,
})

export class EclInpageNavigationItem {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() path: string;

  getClass(): string {
    return [
      `ecl-inpage-navigation__item`,
      `sc-ecl-inpage-navigation-${this.theme}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <li class={this.getClass()}>
        <ecl-link
          path={this.path}
          style-class={`ecl-inpage-navigation__link sc-ecl-inpage-navigation-${this.theme}`}
          data-ecl-inpage-navigation-link
          >
            <slot></slot>
          </ecl-link>
      </li>
    );
  }
}