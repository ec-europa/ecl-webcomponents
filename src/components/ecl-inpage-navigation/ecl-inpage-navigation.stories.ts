import { loremIpsum } from 'lorem-ipsum';

const lorem = loremIpsum({ count: 25 });

export default {
  title: 'Components/Navigation/inpage-navigation',
};

const Template = args =>
`<ecl-grid container>
  <ecl-grid row data-ecl-inpage-navigation-container>
    <ecl-grid columns=3>
      <ecl-inpage-navigation
        inpage-title="Page contents"
        inpage-id="inpage-id"
        theme="${args.theme}"
        ecl-script
      >
        <ecl-inpage-navigation-item
          path="#inline-nav-1"
          theme="${args.theme}"
        >
          Heading 1
        </ecl-inpage-navigation-item>
        <ecl-inpage-navigation-item
          path="#inline-nav-2"
          theme="${args.theme}"
        >
          Heading 2 with a long title going on several lines
        </ecl-inpage-navigation-item>
        <ecl-inpage-navigation-item
          path="#inline-nav-3"
          theme="${args.theme}"
        >
          Heading 3
        </ecl-inpage-navigation-item>
        <ecl-inpage-navigation-item
          path="inline-nav-4"
          theme="${args.theme}"
        >
          Heading 4
        </ecl-inpage-navigation-item>
      </ecl-inpage-navigation>
    </ecl-grid>
    <ecl-grid columns=9 breakpoint="l">
      <ecl-text tag="h2" level="2" item-id="inline-nav-1">Heading 1</ecl-text>
      <ecl-text tag="p" size="m">${lorem}</ecl-text>
      <ecl-text tag="p" size="m">${lorem}</ecl-text>
      <ecl-text tag="h2" level="2" item-id="inline-nav-2">Heading 2</ecl-text>
      <ecl-text tag="p" size="m">${lorem}</ecl-text>
      <ecl-text tag="p" size="m">${lorem}</ecl-text>
      <ecl-text tag="h2" level="2" item-id="inline-nav-3">Heading 3</ecl-text>
      <ecl-text tag="p" size="m">${lorem}</ecl-text>
      <ecl-text tag="h2" level="2" item-id="inline-nav-4">Heading 4</ecl-text>
      <ecl-text tag="p" size="m">${lorem}</ecl-text>
    </ecl-grid>
  </ecl-grid>
</ecl-grid>`;

export const InpageNavigation = Template.bind({});
InpageNavigation.storyName = 'default';