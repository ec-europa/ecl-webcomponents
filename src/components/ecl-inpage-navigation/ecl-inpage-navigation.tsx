import { Component, h, Prop, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-inpage-navigation',
  styleUrls: {
    ec: './build/styles/ecl-inpage-navigation-ec.css',
    eu: './build/styles/ecl-inpage-navigation-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclInpageNavigation {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() eclScript: boolean = false;
  @Prop() inpageTitle: string;
  @Prop() inpageId: string;

  getClass(): string {
    return [
      `ecl-inpage-navigation`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    const items = this.el.querySelectorAll('.ecl-inpage-navigation__item');
    if (items) {
      items.forEach((item) => {
        const link = item.querySelector('.ecl-inpage-navigation__link');
        if (link) {
          link.setAttribute('data-ecl-inpage-navigation-link', 'data-ecl-inpage-navigation-link');
          item.innerHTML = '';
          item.appendChild(link);
        }
      });
      this.el.querySelector('.ecl-inpage-navigation__list').innerHTML = '';
      this.el.querySelector('.ecl-inpage-navigation__list').append(...items);
    }

    if (this.eclScript) { 
      const src = getAssetPath('./build/scripts/ecl-inpage-navigation-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const inpageNavigation = new ECL.InpageNavigation(this.el.firstElementChild);
        inpageNavigation.init();
      };

      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <nav
        class={this.getClass()}
        data-ecl-inpage-navigation
        aria-labelledby={this.inpageId}
      >
        <div
          class="ecl-inpage-navigation__title"
          id={this.inpageId}
        >
          {this.inpageTitle}
        </div>
        <div class="ecl-inpage-navigation__body">
          <button
            class={`ecl-inpage-navigation__trigger sc-ecl-inpage-navigation-${this.theme}`}
            id={`${this.inpageId}-trigger`}
            aria-expanded="false"
            aria-controls="ecl-inpage-navigation-list"
            data-ecl-inpage-navigation-trigger
            aria-label=""
          >
            <span
              class="ecl-inpage-navigation__trigger-current"
              data-ecl-inpage-navigation-trigger-current
            ></span>
            <ecl-icon
              icon="corner-arrow"
              size="s"
              transform="rotate-180"
              style-class="ecl-inpage-navigation__trigger-icon"
            ></ecl-icon>
          </button>
          <ul
            class="ecl-inpage-navigation__list"
            data-ecl-inpage-navigation-list
            id={`${this.inpageId}-list`}
          >
            <slot></slot>
          </ul>
        </div>
      </nav>
    );
  }
}