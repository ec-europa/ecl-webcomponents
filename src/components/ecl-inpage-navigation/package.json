{
  "name": "@ecl/web-components-inpage-navigation",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.12.0",
  "description": "ECL inpage navigation",
  "main": "ecl-inpage-navigation.tsx",
  "private": true,
  "scripts": {
    "build:scripts": "cross-env ecl-builder scripts",
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.13.0",
    "@ecl/vanilla-component-inpage-navigation": "3.13.0",
    "@ecl/vanilla-layout-grid": "3.13.0",
    "cross-env": "7.0.3",
    "lorem-ipsum": "2.0.8"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
