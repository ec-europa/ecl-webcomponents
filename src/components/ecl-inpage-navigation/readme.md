# ecl-inpage-navigation



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `path`       | `path`        |             | `string` | `undefined` |
| `styleClass` | `style-class` |             | `string` | `undefined` |
| `theme`      | `theme`       |             | `string` | `'ec'`      |


## Dependencies

### Depends on

- [ecl-link](../ecl-link)

### Graph
```mermaid
graph TD;
  ecl-inpage-navigation-item --> ecl-link
  ecl-link --> ecl-icon
  style ecl-inpage-navigation-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
