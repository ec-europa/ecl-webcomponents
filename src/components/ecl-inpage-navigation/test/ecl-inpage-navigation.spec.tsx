import { newSpecPage } from '@stencil/core/testing';
import { EclInpageNavigation } from '../ecl-inpage-navigation';
import { EclInpageNavigationItem } from '../ecl-inpage-navigation-item';

describe('ecl-inpage-navigation', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclInpageNavigation, EclInpageNavigationItem],
      html: `<ecl-inpage-navigation
              inpage-title="Page contents"
              inpage-id="inpage-id"
              theme="ec"
              ecl-script
              with-utils
            >
              <ecl-inpage-navigation-item
                path="#inline-nav-1"
              >
                Heading 1
              </ecl-inpage-navigation-item>
              <ecl-inpage-navigation-item
                path="#inline-nav-2"
              >
                Heading 2 with a long title going on several lines
              </ecl-inpage-navigation-item>
              <ecl-inpage-navigation-item
                path="#inline-nav-3"
              >
                Heading 3
              </ecl-inpage-navigation-item>
              <ecl-inpage-navigation-item
                path="inline-nav-4"
              >
                Heading 4
              </ecl-inpage-navigation-item>
            </ecl-inpage-navigation>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
