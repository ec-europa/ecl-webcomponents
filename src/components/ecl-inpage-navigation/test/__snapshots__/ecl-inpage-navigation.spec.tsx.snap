// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-inpage-navigation renders 1`] = `
<ecl-inpage-navigation ecl-script="" inpage-id="inpage-id" inpage-title="Page contents" theme="ec">
  <!---->
  <nav aria-labelledby="inpage-id" class="ecl-inpage-navigation" data-ecl-inpage-navigation>
    <div class="ecl-inpage-navigation__title" id="inpage-id">
      Page contents
    </div>
    <div class="ecl-inpage-navigation__body">
      <button aria-controls="ecl-inpage-navigation-list" aria-expanded="false" aria-label="" class="ecl-inpage-navigation__trigger sc-ecl-inpage-navigation-ec" data-ecl-inpage-navigation-trigger id="inpage-id-trigger">
        <span class="ecl-inpage-navigation__trigger-current" data-ecl-inpage-navigation-trigger-current></span>
        <ecl-icon icon="corner-arrow" size="s" style-class="ecl-inpage-navigation__trigger-icon" transform="rotate-180"></ecl-icon>
      </button>
      <ul class="ecl-inpage-navigation__list" data-ecl-inpage-navigation-list id="inpage-id-list">
        <li class="ecl-inpage-navigation__item sc-ecl-inpage-navigation-ec">
          <ecl-link data-ecl-inpage-navigation-link path="#inline-nav-1" style-class="ecl-inpage-navigation__link sc-ecl-inpage-navigation-ec">
            Heading 1
          </ecl-link>
        </li>
        <li class="ecl-inpage-navigation__item sc-ecl-inpage-navigation-ec">
          <ecl-link data-ecl-inpage-navigation-link path="#inline-nav-2" style-class="ecl-inpage-navigation__link sc-ecl-inpage-navigation-ec">
            Heading 2 with a long title going on several lines
          </ecl-link>
        </li>
        <li class="ecl-inpage-navigation__item sc-ecl-inpage-navigation-ec">
          <ecl-link data-ecl-inpage-navigation-link path="#inline-nav-3" style-class="ecl-inpage-navigation__link sc-ecl-inpage-navigation-ec">
            Heading 3
          </ecl-link>
        </li>
        <li class="ecl-inpage-navigation__item sc-ecl-inpage-navigation-ec">
          <ecl-link data-ecl-inpage-navigation-link path="inline-nav-4" style-class="ecl-inpage-navigation__link sc-ecl-inpage-navigation-ec">
            Heading 4
          </ecl-link>
        </li>
      </ul>
    </div>
  </nav>
</ecl-inpage-navigation>
`;
